src/agni/client/communication/MessageSender.java

```diff
@@ -1,12 +1,17 @@
 package agni.client.communication;
 
 public class MessageSender {
-	
-	String ip;
-	int port;
-	
-	public MessageSender(){
-		
-	}
+
+    private String ip;
+    private int tcpSocket;
+    private int port;
+
+    public MessageSender() {
+
+    }
+
+    public void sendMessage(String message) {
```

Comment:
This type should be `byte[]` so then the the `MessageSender` doesn't need to turn strings into bytes. This will be easier to test. The `ActionHandlers` should do all the byte conversion.